src/ResourceManager/Sql/Commands.Sql/VirtualNetworkRule/Cmdlet/NewAzureSqlServerVirtualNetworkRule.cs

```diff
@@ -0,0 +1,105 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Rest.Azure;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.VirtualNetworkRule.Cmdlet
+{
+    /// <summary>
+    /// Defines the New-AzureRmSqlServerVirtualNetworkRule cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureRmSqlServerVirtualNetworkRule",
+        ConfirmImpact = ConfirmImpact.Low, SupportsShouldProcess = true)]
+    public class NewAzureSqlServerVirtualNetworkRule : AzureSqlServerVirtualNetworkRuleCmdletBase
+    {
+        /// <summary>
+        /// Azure Sql Server Virtual Network Rule Name.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "Azure Sql Server Virtual Network Rule Name.")]
+        [ValidateNotNullOrEmpty]
+        public string VirtualNetworkRuleName { get; set; }
+
+        /// <summary>
+        /// The Virtual Network Subnet Id
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The Virtual Network Subnet Id that specifies the Microsoft.Network details")]
+        [ValidateNotNull]
+        public string VirtualNetworkSubnetId { get; set; }
```

Comment:
From the customer's perspective, they will get this entire id the minute they create the vnet using networking cmdlets. So it will be a copy paste for them. If we separate it out, they would have to manually separate out parts of the id and then pass it to us.